src/tests/ftest/container/auto_oc_selection.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020-2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from apricot import TestWithServers
+from test_utils_container import TestContainer
+from avocado.core.exceptions import TestFail
+
+
+class AutoOCSelectionTest(TestWithServers):
+    """Automatic object class selection test.
+
+    Test Class Description:
+    Create a container with different RF and OC combinations. We expect some
+    combinations to fail due to the restriction set by DAOS. (This is more like
+    auto OC "restriction" rather than auto OC "selection".)
+
+    :avocado: recursive
+    """
+    def test_oc_restrictions(self):
+        """JIRA ID: DAOS-7595
+
+        Test Description:
+        Create a container with RF 0 to 4 - OC S1, RP_2G1, RP_3G1, RP_4G1, and
+        RP_5G1.
+
+        Use Cases:
+        Create a pool. Then create each container and verify against its
+        failure expected state.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=vm
+        :avocado: tags=container,oc_restrictions
+        """
+        self.add_pool()
+
+        # (Redundancy Factor, Object Class, Failure Expected)
+        prop_oclasses = [
+            ("rf:0", "S1", False),
+            ("rf:0", "RP_2G1", False),
+            ("rf:0", "RP_3G1", False),
+            ("rf:0", "RP_4G1", False),
+            ("rf:0", "RP_5G1", False),
+            ("rf:1", "S1", True),
+            ("rf:1", "RP_2G1", False),
+            ("rf:1", "RP_3G1", False),
+            ("rf:1", "RP_4G1", False),
+            ("rf:1", "RP_5G1", False),
+            ("rf:2", "S1", True),
+            ("rf:2", "RP_2G1", True),
+            ("rf:2", "RP_3G1", False),
+            ("rf:2", "RP_4G1", False),
+            ("rf:2", "RP_5G1", False),
+            ("rf:3", "S1", True),
+            ("rf:3", "RP_2G1", True),
+            ("rf:3", "RP_3G1", True),
+            ("rf:3", "RP_4G1", False),
+            ("rf:3", "RP_5G1", False),
+            ("rf:4", "S1", True),
+            ("rf:4", "RP_2G1", True),
+            ("rf:4", "RP_3G1", True),
+            ("rf:4", "RP_4G1", True),
+            ("rf:4", "RP_5G1", False)
+        ]
+        self.container = []
+        errors = []
+
+        # Create a container for each combination.
+        for prop_oclass in prop_oclasses:
+            properties = prop_oclass[0]
+            oclass = prop_oclass[1]
+            failure_expected = prop_oclass[2]
+
+            cont = TestContainer(
+                pool=self.pool, daos_command=self.get_daos_command())
+            cont.get_params(self)
+            cont.properties.update(properties)
+            cont.oclass.update(oclass)
+
+            try:
+                self.log.debug(
+                    "Create container with RF = %s, OC = %s", properties,
+                    oclass)
+                cont.create()
```

Comment:
Suggestion/Optional - Could we use `TestWithServers.get_container()`, but with added params for `properties` and `oclass` so other tests can benefit?

Reply:
Done